build_runner_core/lib/src/generate/build_directory.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+class BuildDirectory {
+  final String directory;
+  final OutputLocation outputLocation;
+  BuildDirectory(this.directory, this.outputLocation);
```

Comment:
nit: move `outputLocation` to an optional named arg since its optional, and document the behavior for when its null